### PR TITLE
New sdk for Azure Blueprint

### PR DESCRIPTION
**Pull request by haitch**
## Description
Add new SDK for Azure Blueprint RP.
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/3855 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.Blueprint</PackageId>
+    <Description>Provides Microsoft Azure Blueprint management operations including the ability to create, update and delete Blueprint definition and assignment.</Description>
+    <Version>0.1.0-Preview</Version>
```
Comment (dsgouda) [inline src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: Preview versions must start from `0.9.0-preview`

File: src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <!-- Remove the Preview description in PackageReleaseNotes once the preview is removed.-->
+    <AssemblyTitle>Microsoft Azure Blueprint Management Library</AssemblyTitle>
+    <AssemblyName>Microsoft.Azure.Management.Blueprint</AssemblyName>
+    <PackageTags>Blueprint;</PackageTags>
+    <PackageReleaseNotes>
```
Comment (dsgouda) [inline src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: Please add some relevant information in `PackageReleaseNotes`
An example [here](https://github.com/Azure/azure-sdk-for-net/blob/53b16a9d7954076f9002013846c4c2840cebf92b/src/SDKs/DevSpaces/Management.DevSpaces/Microsoft.Azure.Management.DevSpaces.csproj#L12)

File: src/SDKs/Blueprint/Blueprint.sln [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Net40-Release|Any CPU.Build.0 = Release|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Net45-Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Net45-Debug|Any CPU.Build.0 = Debug|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Net45-Release|Any CPU.ActiveCfg = Release|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Net45-Release|Any CPU.Build.0 = Release|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Portable-Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Portable-Debug|Any CPU.Build.0 = Debug|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Portable-Release|Any CPU.ActiveCfg = Release|Any CPU
+		{E93AD1AE-BC72-45C9-90EB-878D45169599}.Portable-Release|Any CPU.Build.0 = Release|Any CPU
+		{6E64088C-3182-4E50-9450-2365508135F7}.Net40-Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (dsgouda) [inline src/SDKs/Blueprint/Blueprint.sln]: The only allowed configs are net452 and netstandard, please remove all other configs here

File: src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj (hunk 2 above)
Comment (dsgouda) [reply src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: Please add `AssemblyInfo.cs` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Logic/Management.Logic/Properties/AssemblyInfo.cs) with versioning info

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please commit any txt files generated during code generation

**Comment by dsgouda** [comment]
Please squash commits into a single commit and update the PR

**Comment by haitch** [comment]
> Please squash commits into a single commit and update the PR

we can do a squash merge, right?

File: src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj (hunk 2 above)
Comment (haitch) [reply src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: ✅

Comment (haitch) [reply src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: ✅

File: src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj (hunk 1 above)
Comment (haitch) [reply src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj]: ✅

**Comment by dsgouda** [comment]
@haitch not on this PR. You will have to recreate the branch with a single commit and force push

**Comment by haitch** [comment]
> @haitch not on this PR. You will have to recreate the branch with a single commit and force push

✅

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**